infra/terraform/modules/kubernetes_access_control/data.tf

```diff
@@ -0,0 +1,129 @@
+data "aws_iam_policy_document" "kubernetes_assume_role_policy" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+    principals {
+      type        = "Service"
+      identifiers = ["ec2.amazonaws.com"]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "kubernetes_nodes_aws_iam_role_policy" {
+  statement {
+    effect    = "Allow"
+    actions   = ["ec2:Describe*"]
+    resources = ["*"]
+  }
+
+  statement {
+    effect  = "Allow"
+    actions = [
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+      "logs:DescribeLogStreams",
+      "logs:DescribeLogGroups"
+    ]
+    resources = ["arn:aws:logs:*:*:*"]
+  }
+
+  statement {
+    effect  = "Allow"
+    actions = [
```

Comment:
I'm just now realised that k8s is using ECR when the nodes are EC2 instances...

Reply:
The ECR permissions come from Kubernetes itself I think, to support people using ECR to host their Docker containers without having to handle ECR creds within the cluster. We have a couple of extra permissions in there to allow pushing images, as our users' app containers are pushed to ECR.